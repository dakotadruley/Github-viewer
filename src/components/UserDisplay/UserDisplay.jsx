import React from 'react';
import PropTypes from 'prop-types';

const UserDisplay = ({ name, avatar_url, followers, following, html_url }) => (
  <section>
    <h2>{name}</h2>
    <img src={avatar_url} alt={name} />
    <a href={html_url}>GitHub</a>
    <p>{followers}</p>
    <p>{following}</p>
  </section>
);

UserDisplay.propTypes = {
  name: PropTypes.string.isRequired,
  avatar_url: PropTypes.string.isRequired,
  html_url: PropTypes.string.isRequired,
  followers: PropTypes.string.isRequired,
  following: PropTypes.string.isRequired
};

export default UserDisplay;
