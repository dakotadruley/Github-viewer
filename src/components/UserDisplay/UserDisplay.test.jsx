import React from 'react';
import { shallow } from 'enzyme';
import UserDisplay from './UserDisplay.jsx';

describe('UserDisplay', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<UserDisplay 
      name="dakota druley" 
      link="URL"
      followers="10" 
      following="15"/>);
    expect(wrapper).toMatchSnapshot();
  });
});
