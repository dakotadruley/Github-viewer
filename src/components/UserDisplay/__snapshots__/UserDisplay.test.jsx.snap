// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDisplay matches a snapshot 1`] = `
<section>
  <h2>
    dakota druley
  </h2>
  <img
    alt="dakota druley"
  />
  <a>
    GitHub
  </a>
  <p>
    10
  </p>
  <p>
    15
  </p>
</section>
`;
